### PR TITLE
cli: change exit functions to never return

### PR DESCRIPTION
**Pull request by mzabaluev**
## Description

A drive-by cleanup made while working on #1834

______

### PR author checklist:
- ~Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).~
- ~Added tests: integration (for Hermes) or unit/mock tests (for modules).~ 
- ~Linked to GitHub issue.~
- [x] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [x] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by mzabaluev** [comment]
md-link-check failure looks spurious.

**Comment by romac** [comment]
> md-link-check failure looks spurious.

It is, just restarted the job

File: relayer-cli/src/conclude.rs [hunk 1]
```diff
@@ -208,7 +205,7 @@ impl Output {
     }
 
     /// Exits from the process with the current output. Convenience wrapper over `exit_with`.
-    pub fn exit(self) {
+    pub fn exit(self) -> ! {
```
Comment (romac) [inline relayer-cli/src/conclude.rs:208]: 🥇

**Review by romac — APPROVED** [review]

**Review by adizere — COMMENTED** [review]
💘

**Pull request merged**